src/node_http_parser.cc

```diff
@@ -644,6 +866,26 @@ class Parser : public AsyncWrap, public StreamListener {
     args.GetReturnValue().Set(ret);
   }
 
+  static void Duration(const FunctionCallbackInfo<Value>& args) {
+    Parser* parser;
+    ASSIGN_OR_RETURN_UNWRAP(&parser, args.Holder());
+
+    if (parser->last_message_start_ == 0) {
+      args.GetReturnValue().Set(0);
+      return;
+    }
+
+    uint64_t duration = (uv_hrtime() - parser->last_message_start_) / 1000000;
```

Comment:
Is this supposed to be an integer given that we are casting it into a double in the next line? Can't we just assign this to a `double` data type and divide with `1e6` instead?

Reply:
👍